lib/rexml/parsers/baseparser.rb

```diff
@@ -274,8 +274,8 @@ def pull_event
             return [ :elementdecl, @source.match( ELEMENTDECL_PATTERN, true )[1] ]
 
           when ENTITY_START
-            match = @source.match( ENTITYDECL, true ).to_a.compact
-            match[0] = :entitydecl
+            match = @source.match( ENTITYDECL, true )
+            match = match.nil? ? [:entitydecl] : [:entitydecl, *match.captures.compact.reject(&:empty?)]
```

Comment:
Hmm. This assumes that `match` is `StringScanner`.
How about returning `@scanner.captures` instead of `@scanner` by `@source.match`?

Reply:
I believe that https://github.com/ruby/strscan/pull/72 needs to be merged in order to use `@scanner.captures`.

Reply:
Merged and released.

Reply:
Changed `source#match?` to return `scanner.captures` instead of `@scanner`.

Reply:
Add compact option to `@source.match` with 8bc895562c85e4689544277f5a8168a1822b0b50

Improve processing speed by returning `@scanner.captures.compact` if `@compact=true` and `@scanner` if `compact=false`.

Reply:
Added `match?` and removed `compact` option in 50b305711d1a4401975843a88cd6315ea2d1f9df.

Reply:
Removed `Source#match?` and return `@scanner` in Source#match.